src/vrct/vrctFs/vrct.go

```diff
@@ -0,0 +1,280 @@
+package vrctFs
+
+import (
+	"fmt"
+	"gopkg.in/mgo.v2/bson"
+	"os"
+	"slices"
+	"sort"
+	"strings"
+)
+
+const virtualFSPathPrefix = "/tmp/spito-vrct/fs"
+
+type FsVRCT struct {
+	virtualFSPath  string
+	fsRequirements []FsRequirement
+}
+
+func (v *FsVRCT) EnsureInitialized() error {
+	if v.fsRequirements == nil {
+		v.fsRequirements = make([]FsRequirement, 0)
+	}
+
+	if v.virtualFSPath != "" {
+		return nil
+	}
+
+	err := os.MkdirAll(virtualFSPathPrefix, os.ModePerm)
+	if err != nil {
+		return err
+	}
+
+	dir, err := os.MkdirTemp(virtualFSPathPrefix, "")
+	if err != nil {
+		return err
+	}
+
+	v.virtualFSPath = dir
+
+	return nil
+}
+
+func (v *FsVRCT) checkRequirements() (bool, *FsRequirement) {
+	for _, req := range v.fsRequirements {
+		if req.checkRequirement() == false {
+			return false, &req
+		}
+	}
+
+	return true, nil
+}
+
+func (v *FsVRCT) InnerValidate() error {
+	for _, requirement := range v.fsRequirements {
+		if !requirement.checkRequirement() {
+			return fmt.Errorf("requirement %v is not met", requirement.ruleStackTrace)
+		}
+	}
+
+	return nil
+}
+
+func (v *FsVRCT) Apply() error {
+	if err := v.InnerValidate(); err != nil {
+		return err
+	}
+
+	mergeDir, err := os.MkdirTemp("/tmp", "spito-fs-vrct-merge")
+	if err != nil {
+		return err
+	}
+
+	if err := v.mergeFsChanges(v.virtualFSPath, mergeDir); err != nil {
+		return err
+	}
+
+	// TODO login after merge (this time merge to real fs)
+
+	return nil
+}
+
+func (v *FsVRCT) mergeFsChanges(prototypesDirPath, destPath string) error {
```

Comment:
What type is this ```prototypesDirPath```?

Reply:
String you are probably confused because it is strange golang syntax where you can e.g. declare such function:
`func test(x string, y, z int)` (in this example x is string, while y and z both int)